cmake/cc_rules.cmake

```diff
@@ -116,6 +116,14 @@ function(firebase_ios_add_test target)
 
   firebase_ios_set_common_target_options(${target} ${flag_OPTIONS})
 
+  # Set a preprocessor define so that tests can distinguish between having been
+  # built by Xcode vs. cmake.
+  target_compile_definitions(
+    ${target}
+    PRIVATE
+    FIREBASE_TESTS_BUILT_BY_CMAKE
+  )
```

Comment:
This repository can be compiled via two completely separate and independent build chains: cmake and xcode. Both are configured completely independently. Snappy is _only_ available when compiled via cmake. When compiled via xcode then snappy support is _not_ compiled in.

The unit tests in `leveldb_snappy_test.cc` check that snappy support _is_ available when compiled via cmake and that snappy support is _not_ available when compiled via xcode. That test file uses this preprocessor variable `FIREBASE_TESTS_BUILT_BY_CMAKE` to determine which snappy availability to verify, since that preprocessor variable will NOT be defined when compiled via xcode.